services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/grpc/LogicalPortSpec.groovy

```diff
@@ -0,0 +1,106 @@
+    @Shared
+    Integer switchPort
+    @Shared
+    Integer switchLogicalPort
+
+    def setupOnce() {
+        requireProfiles("hardware")
+        def nFlowSwitch = northbound.activeSwitches.find { it.description =~ /NW[0-9]+.[0-9].[0-9]/ }
+        def pattern = /(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\-){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)/
+        switchIp = (nFlowSwitch.address =~ pattern)[0].replaceAll("-", ".")
+        switchPort = northbound.getPorts(nFlowSwitch.switchId).find { it.state[0] == "LINK_DOWN" }.portNumber
+        switchLogicalPort = 1100 + switchPort
+    }
+
+    def "Able to create/read/delete logicalport"() {
+        /**the update action is not working(issue on a Noviflow switch side)*/
+        when: "Create logical port"
+        grpc.createLogicalPort(switchIp, new LogicalPortDto([switchPort], switchLogicalPort))
+
+        then: "Able to get the created logical port"
+        def r = grpc.getSwitchLogicalPortConfig(switchIp, switchLogicalPort)
+        r.logicalPortNumber == switchLogicalPort
+        r.name == "novi_lport" + switchLogicalPort.toString()
+        r.portNumbers[0] == switchPort
+
+        and: "The created port is exist in a list of all logical port"
+        grpc.getSwitchLogicalPorts(switchIp).contains(r)
+
+        //        TODO(andriidovhan): add update action
+        //        and: "able to edit the created logical port"
+        when: "Try to delete the created logical port"
+        def r1 = grpc.deleteSwitchLogicalPort(switchIp, switchLogicalPort)
+
+        then: "Logical port is deleted"
+        r1.deleted
+
+        when: "Try to get the deleted logical port"
+        grpc.getSwitchLogicalPortConfig(switchIp, switchLogicalPort)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.statusCode == HttpStatus.NOT_FOUND
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Provided logical port does not exist."
+        Boolean testIsCompleted = true
+
+        cleanup: "Remove created port"
+        if (!testIsCompleted) {
+            grpc.deleteSwitchLogicalPort(switchIp, switchLogicalPort)
+        }
+    }
+
+    @Unroll
+    def "Not able to create logical port with incorrect port number(lPort/sPort): #logicalPortNumber/#portNumber"() {
+        when:
+        "Try to create logical port: $logicalPortNumber/$portNumber"
+        grpc.createLogicalPort(switchIp, new LogicalPortDto([portNumber], logicalPortNumber))
+
+        then: "Human readable error is returned."
+        def exc = thrown(HttpClientErrorException)
+        exc.statusCode == HttpStatus.BAD_REQUEST
+        exc.responseBodyAsString.to(MessageError).errorMessage == errorMessage
+
+        where:
+        logicalPortNumber | portNumber | errorMessage
+        99                | switchPort | "Valid logicalportno range is 100 to 63487."
+        63488             | switchPort | "Valid logicalportno range is 100 to 63487."
```

Comment:
This error  is also taken from official documentaion for Noviflow switches
and it can't be changed
`Error	185:	Valid logicalportno	range	is	100	to	63487`